### ISSUE TITLE
RAPL as perf event

### ISSUE THREAD

**Issue by tilsche** [tud-zih-energy/lo2s#128]
Doesn't work (tried with cores/pkg)

```
$ lo2s -a -E power/energy-cores/
[642010394386][pid: 3568][tid: 140581833051008][ERROR]: perf_event_open for counter failed
[642010562825][pid: 3568][tid: 140581833051008][ERROR]: failed to add counter 'power/energy-cores/': Invalid argument
[642010689808][pid: 3568][tid: 140581833051008][ERROR]: opening 1 counters at once might exceed the hardware limit of simultaneously openable counters.
[642033308460][pid: 3568][tid: 140581833051008][FATAL]: Aborting: Invalid argument
```
This is somewhat consistent with `perf record`:
```
$ perf record -e '{cycles,power/energy-pkg/}:S' -- sleep 1
Error:
The sys_perf_event_open() syscall returned with 22 (Invalid argument) for event (power/energy-pkg/).
/bin/dmesg may provide additional information.
```

Clearly `perf record -e power/energy-pkg/` can't work - but the group thing *should work*.

However, `perf stat` works fine. 

Related:
https://linux-perf-users.vger.kernel.narkive.com/dIN2TKRY/recording-rapl-power-events

I haven't yet tried an older version of `lo2s` before the group recording.

 Given that perf doesn't work, this may very well be a bug / limitation of the kernel itself.

**Comment by cvonelm** [comment]
The pre group recording variant should work, as the problem is that only hardware events that share a PMU can be grouped together.

**Comment by tilsche** [comment]
oh... Is there a way to support it with the current architecture without complicating everything?

**Comment by cvonelm** [comment]
Im pretty afraid the answer is no, while ```if(EINVAL) { retry outside of group }``` should be easy enough there is no way we will get useful timestamps without a metric leader

**Comment by cvonelm** [comment]
Thinking about this longer than 3.5 seconds, there is a way to do this.

The trick here is using an extra Monitor which does not use MMAPs to read perf events, but the classic way of directly read()-ing the fd, and using lo2s::time::now() to get timestamps, like we use it for x86_energy:

```cpp
 void Monitor::monitor(int fd)
 {
     (void)fd;
     metric_event_.timestamp(time::now());
     metric_event_.raw_values()[0] = read(fd);

     otf2_writer_.write(metric_event_);
 }
```

Because this is different from "normal" metric, i would like to introduce a new switch for this.

For this I would need the use case. Do we want this solely for percentage spent in C0 (mperf/aperf or aperf/mperf, I have two different links that say different things) or is there more to it?

**Comment by cvonelm** [comment]
Closed in favour of the broader #158 issue
